### PR TITLE
[search] various improvements/fixes around datetimes

### PR DESCRIPTION
**Pull request by dcramer**
- add activeSince to search
- add firstSeen to search
- add lastSeen to search
- add firstRelease to search
- fix usage of active_at in seen by checks
- populate active_at on new groups
- support +/- date helpers in range queries (e.g. ``age:<=-1d``)

### CODE REVIEW

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def get_attrs(self, item_list, user):
 
         result = {}
         for item in item_list:
-            active_date = item.active_at or item.last_seen
+            active_date = item.active_at or item.first_seen
```
Comment (dcramer) [inline src/sentry/api/serializers/models/group.py]: this seemingly was broken, according to my super old docstring on what active_at is supposed to mean

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -556,6 +556,7 @@ def save(self, project, raw=False):
             'level': level,
             'last_seen': date,
             'first_seen': date,
+            'active_at': date,
```
Comment (dcramer) [inline src/sentry/event_manager.py]: this also wasnt populating -- i couldnt find any reason why, but according to docstring it seems correct

File: src/sentry/search/django/backend.py [hunk 3]
```diff
@@ -169,6 +171,20 @@ def _build_queryset(self, project, query=None, status=None, tags=None,
                     params['last_seen__lt'] = last_seen_to
             queryset = queryset.filter(**params)
 
+        if active_at_from or active_at_to:
```
Comment (dcramer) [inline src/sentry/search/django/backend.py]: these likely aren't indexed well, so we might want to move this into a subquery to limit the depth of the search

File: src/sentry/search/utils.py [hunk 4]
```diff
@@ -274,7 +282,7 @@ def parse_query(project, query, user):
                 results['bookmarked_by'] = parse_user_value(value, user)
             elif key == 'subscribed':
                 results['subscribed_by'] = parse_user_value(value, user)
-            elif key == 'first-release':
+            elif key in ('first-release', 'firstRelease'):
```
Comment (TheSavior) [inline src/sentry/search/utils.py]: What causes the different word cases?

Comment (dcramer) [inline src/sentry/search/utils.py]: @TheSavior mostly thinking around standardization. We have some _ and some -, and most things we use camelCase when possible (or dot to separation subsets).

File: src/sentry/api/serializers/models/group.py (hunk 1 above)
Comment (mattrobenolt) [inline src/sentry/api/serializers/models/group.py]: :+1:

File: src/sentry/search/django/backend.py (hunk 3 above)
Comment (mattrobenolt) [inline src/sentry/search/django/backend.py]: These are indexed.

**Review by mattrobenolt — APPROVED** [review]

**Comment by TheSavior** [comment]
Is there anything left on this?

**Comment by TheSavior** [comment]
Great, thanks! How frequently do changes typically get deployed to sentry.io? I'm checking this search periodically. :) `is:unresolved active_at:<=-24h`

**Comment by dcramer** [comment]
It's live but you'll want activeSince

**Comment by TheSavior** [comment]
Ah, I read the changelog backwards then. Thanks!

**Pull request merged**